common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -93,68 +93,40 @@ public static String modifyMultipleMatchKeyword(String query){
                         List<Predicate> synonymConceptPredicate = new ArrayList<>();
                         Expression<Long> conceptIdCheck = root.get("conceptId");
                         synonymConceptPredicate.add(conceptIdCheck.in(synonymConceptIds));
-                        predicates.add(criteriaBuilder.or(standardConceptPredicates.toArray(new Predicate[0])));
-                        if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS) || standardConceptFilter.equals(StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH)) {
-                                predicates.add(criteriaBuilder.or(synonymConceptPredicate.toArray(new Predicate[0])));
-                        } else if (standardConceptFilter.equals(StandardConceptFilter.NON_STANDARD_CONCEPTS)) {
-                            predicates.add(criteriaBuilder.or(
-                                            criteriaBuilder.or(criteriaBuilder.isNull(root.get("standardConcept"))),
-                                            criteriaBuilder.and(nonStandardConceptPredicates.toArray(new Predicate[0]))));
-                        }
+                        predicates.add(criteriaBuilder.or(synonymConceptPredicate.toArray(new Predicate[0])));
                     }
-                    else{
-                        List<Predicate> conceptCodeIDName = new ArrayList<>();
-                        Expression<Double> matchExp = null;
-                        if(!Strings.isNullOrEmpty(query)){
-                            final String keyword = modifyMultipleMatchKeyword(query);
-                            conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptCode"),
+                    else {
```

Comment:
This is all much cleaner.  One issue we may have is that if a code or conceptid happens to be in the name or synonym string, the name/synonym takes precedence here.  I think we  probably still have that problem in the master now. 

The problem is we can't tell if something is a code or concept_id until we match it.  
The logic i think needs to be : 
1) Search code / concept id  to get concept ids
2) If not code match then search name synonm to get concept ids 
3) Call the service to apply the predicate filters with list of concept ids.

Reply:
Also, the standardConcept_orCodeId filter seems to not get applied anymore.   I think the solution is to have a separate findConceptIdByCodeOrId method ... And apply that first like you do for the synonms .. Or do it inside the service searchConcepts .. You could do both inside the service searchConcepts right ? rename your variable in conceptServiceSearchConcepts to "conceptIds" instead of synonmy  ... Of course then you know the match type already before you use the service. So that will have to be handled.

Let me put in some pseudocode and commit it in this branch and you can try to implement it.